ideascube/mixins.py

```diff
@@ -72,10 +74,10 @@ def _set_available_tags(self, context):
 
     def get_context_data(self, **kwargs):
         context = super(FilterableViewMixin, self).get_context_data(**kwargs)
-        for key in ('q', 'kind', 'lang'):
+        for key in ('q', 'kind', 'lang', 'source'):
             context[key] = self.request.GET.get(key)
         context['tags'] = self.request.GET.getlist('tags')
-        current_filters = [(k, context[k]) for k in ('kind', 'lang')
+        current_filters = [(k, context[k]) for k in ('kind', 'lang', 'source')
```

Comment:
> It would complicate the for loop with a inner if.

But it would mean less looping, and thus less work for the poor Koombooks.

Anyway, not something for this PR, but in general I find we do a lot of unnecessary looping in many places of the code, and it might pay to try and make an effort at some point in the future on reducing that.